### DISCUSSION TITLE
Rename the `--dumps-dir` CLI options

### DISCUSSION THREAD

**Discussion by Kerollmops** [meilisearch/product#560] [Feedback & Feature Proposal]
The current CLI `--snapshot-dir` option should be pluralized as the `--dumps-dir` option is. The descriptions of this flag are plurals too.

**Comment by gmourier** [comment]
Thanks @Kerollmops

**Comment by dichotommy** [comment]
Hey, hope I'm not too late in saying this, but I would actually suggest the opposite: `--dumps-dir` should become `--dump-dir`, and `--snapshot-dir` should remain unchanged (or at least, not made plural).

Tagging @maryamsulemani97 and @guimachiavelli just in case I've gone temporarily insane 🙃

**Reply by gmourier** [reply]
Thanks, @dichotommy. We will follow the doc team's input on that subject!

**Reply by guimachiavelli** [reply]
I agree with @dichotommy. It sounds more natural to say "snapshot directory", instead of  "a directory of snapshots". The same for dumps: "dump directory" instead of "a directory of dumps".

**Reply by gmourier** [reply]
Perhaps there is a clarification that could help.

The `snapshot-dir` contains only one snapshot at a time, while `dumps-dir` can contain multiple dumps.

**Reply by guimachiavelli** [reply]
So, IMHO, the amount of snapshots/dumps at any given time is not relevant for naming. These settings refer more to a type of folder than to its contents: we have dumps, and we have the dump folder; we have snapshots and the snapshot folder. Using the plural is probably not wrong, but I do feel it's more idiomatic to use the singular.

I think both points become more concrete once we try to pluralise these names: `--snapshots-folder`s or `--dumps-folder`s  sound wrong, whereas `--snapshot-folder`s or `--dump-folder`s flow very easily. Obviously it makes no sense to refer to multiple snapshot folders in a single instance, but I can see us mentioning multiple folders in documentation and support.

**Reply by gmourier** [reply]
Thank you @guimachiavelli !

**Comment by gmourier** [comment]
If I recap so far

- `dumps-dir` should be renamed  to `dump-dir`.
- `MEILI_DUMPS_DIR` should be renamed to `MEILI_DUMP_DIR`.

I looked for plural forms on other `snapshot` and `dump` options, this one seems to be the only one impacted. An extra-look would be appreciated.

Signal @nicolasvienot to have your input on that. Is this something you can manage in an upgrade process that would go from v0.29 to v1.0?

**Reply by nicolasvienot** [reply]
Hey @gmourier,
What do you mean by `in an upgrade process`?
I don't think this would be a big issue for the @meilisearch/cloud-team. Currently, we set `MEILI_DUMPS_DIR` and we can start setting `MEILI_DUMP_DIR` as soon as the engine-team launches a new version with this change.

**Reply by gmourier** [reply]
Hey @nicolasvienot, perfect; thanks for your answer!

I was wondering if you had the flexibility to tell the machines of such versions; "use these env var precisely" to see if it would not be a bblocker on your side.

**Reply by Kerollmops** [reply]
In fact, you could always set both `MEILI_DUMPS_DIR` and `MEILI_DUMP_DIR`, one of them will be ignored depending on the version of Meilisearch 😂

**Reply by curquiza** [reply]
Issue created: https://github.com/meilisearch/meilisearch/issues/3132

**Comment by dureuill** [comment]
So I started working on https://github.com/meilisearch/meilisearch/issues/3132 and it raised a few additional questions that I was to told to direct here:

1. should the default dump dir remain `dumps/` or become `dump/`?
2. the documentation has a [`Dumps destination` section](https://docs.meilisearch.com/learn/configuration/instance_options.html#dumps-destination) (that is being referenced by e.g. the default `config.toml`), should we also change that section to be `Dump destination`, as it is a compound noun also?

Thanks in advance for any answer!

**Reply by gmourier** [reply]
Hey, @meilisearch/docs-team 👋 Any thoughts regarding @dureuill's questions?

Thanks!

**Reply by dichotommy** [reply]
Hey @dureuill 👋🏻 

1. The default dump dir should certainly remain `dumps` as it can contain multiple dump files. `snapshots` is a trickier question, since there will only ever be one snapshot—which FWIW does defeat the purpose of creating a folder. Regardless, it is generally more intuitive for folders to have plural names (e.g. `documents`, `photos`, `snapshots`, `dumps`) so I would recommend keeping both plural.
2. I would agree with this change, and would go further to say that it should be retitled `Dump directory` so that it uses the same language as the config option.

Thanks for your questions 😊

**Reply by dureuill** [reply]
Hey @dichotommy 👋 

Thanks for the answer, so nothing to do for (1), and as we agree about (2) to rename to `Dump directory`, then I will do the PR in the meilisearch/meilisearch repository and amend my meilisearch/specifications PR to use that terminology consistently (e.g. updating the `config.toml` that currently references `Dumps destination`)

**Reply by dureuill** [reply]
the meilisearch/meilisearch PR is [here](https://github.com/meilisearch/meilisearch/pull/3199), and the [specification PR](https://github.com/meilisearch/specifications/pull/204) is up-to-date. I'm not sure when we should merge them though (at the moment the link is broken as expected since the v0.30 documentation has the original "Dumps destination" header)

**Reply by gmourier** [reply]
Hey @dureuill 👋

Thank you for being super active on that. Regarding the specifications, we can review and approve them. Generally, we merge them on the day of the release. (We already have cases where when it is merged too early, we make X PRs to make changes afterward before the release, which is not practical 🤯).

**Reply by Kerollmops** [reply]
We currently create a _snapshots_ directory to put the snapshot in it because it is named the same as the original database i.e. _data.ms_. This way we avoid name conflicts.

Why not changing the default `--snapshot-dir` value to be the current directory and use another name for the generated snapshot like _snapshot.mdb_?

**Comment by gmourier** [comment]
I've changed the title to be focused on the dumps and not the snapshot as initially given

**Reply by dureuill** [reply]
thanks, that was indeed a bit confusing!
